admin-dev/themes/new-theme/js/pages/product/components/combination-modal/History.vue

```diff
@@ -0,0 +1,218 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ *-->
+<template>
+  <div
+    class="card history"
+    @click="preventClose"
+  >
+    <div class="card-header">
+      {{
+        $t("modal.history.editedCombination", {
+          "%editedNb%": combinationsList.length,
+        })
+      }}
+    </div>
+
+    <div class="card-block">
+      <ul
+        class="history-list"
+        v-if="areCombinationsNotEmpty"
+      >
+        <li
+          :class="['history-item', isSelected(combination.id)]"
+          v-for="(combination, key) of paginatedDatas[currentPage - 1]"
+          :key="key"
+          @click="selectCombination(combination)"
+        >
+          {{ combination.title }}
+          <i class="material-icons">edit</i>
+        </li>
+      </ul>
+      <div
+        class="history-empty"
+        v-else
+      >
+        <img :src="emptyImage">
+        <p class="history-empty-tip">
+          {{ $t("modal.history.empty") }}
+        </p>
+      </div>
+    </div>
+    <div
+      class="card-footer"
+      v-if="areCombinationsNotEmpty"
+    >
+      <pagination
+        :pagination-length="14"
+        :datas="combinationsList"
+        @paginated="constructDatas"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+  import ProductEventMap from '@pages/product/product-event-map';
+  import Pagination from '@vue/components/Pagination';
+
+  const CombinationsEventMap = ProductEventMap.combinations;
+
+  export default {
+    name: 'CombinationHistory',
+    data() {
+      return {
+        paginatedDatas: [],
+        currentPage: 1,
+      };
+    },
+    components: {
+      Pagination,
+    },
+    props: {
+      combinationsList: {
+        type: Array,
+        default: () => [],
+      },
+      selectedCombination: {
+        type: Number,
+        required: true,
+      },
+      emptyImage: {
+        type: String,
+        required: true,
+      },
+    },
+    computed: {
+      areCombinationsNotEmpty() {
+        return this.combinationsList.length > 0;
+      },
+    },
+    mounted() {
+      this.$parent.$on(CombinationsEventMap.selectCombination, (id) => {
```

Comment:
I wouldn't have used the event map in this case because it's an internal event in the Vue component, and the parent must use the hardcoded value anyway => `@selectCombination="selectCombination"`
But it's not important anyway ^^